base_decorator.rb

```diff
@@ -0,0 +1,10 @@
+class BaseDecorator < Nameable
```

Comment:
- [ ] Kindly ensure you test your app. You can use the demo code shared with you in [the project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md). I get an error here because you did not import the `nameable` class. Kindly import that.

  ![image](https://user-images.githubusercontent.com/36542986/228779044-cb39dd11-44af-4bdb-a717-724dfcbe3763.png)

Reply:
@AbdifatahYasin1, here is an example:

- Create a new file, like `test.rb`
- Copy this code into it:

  ```
  require_relative './capitalize_decorator.rb'
  require_relative './trimmer_decorator.rb'
  require_relative './person.rb'
  
  person = Person.new(age: 22, name: 'maximilianus')
  p person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  p capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimDecorator.new(capitalizedPerson)
  p capitalizedTrimmedPerson.correct_name
  ```
- Run this command to see the results:

  ```
  ruby test.rb
  ```